src/test/java/org/broadinstitute/barclay/argparser/CommandLinePluginUnitTest.java

```diff
@@ -453,4 +455,26 @@ public void testPluginArgumentNameCollision(){
                 Collections.emptySet());
     }
 
+    @DataProvider(name="includePluginClassTests")
+    Object[][] getIncludePluginClassTests() {
+        return new Object[][] {
+                { TestPluginBase.class, true },
+                { TestPluginWithOptionalArg.class, true },
+                { Integer.class, false }
+        };
+    }
+
+    // Used to test the default implementation of includePluginClass. Provides an explicit implementation of
+    // getPluginBaseClass since that will be called by the mock.
+    private abstract class DescriptorThatDoesntOverrideIncludePluginClass extends CommandLinePluginDescriptor<TestPluginBase> {
+        public Class<?> getPluginBaseClass() { return TestPluginBase.class; }
+    }
+
+    @Test(dataProvider = "includePluginClassTests")
+    public void testDefaultIncludePluginClass(final Class<?> clazz, final boolean expectedInclusion) {
+        final CommandLinePluginDescriptor<?> descriptor =
+                Mockito.mock(DescriptorThatDoesntOverrideIncludePluginClass.class, Mockito.CALLS_REAL_METHODS);
```

Comment:
This is really cool.  I should really learn to use Mockito. Nice suggestion @magicDGS.